### ISSUE TITLE
Request: Add support for horizontal tiling

### ISSUE THREAD

**Issue by mstoeckl** [invoke-ai/InvokeAI#963]
**Is your feature request related to a problem? Please describe.**

InvokeAI currently has an option for "Seamless tiling", which tiles an image so that one can arrange the image in a grid without significant seams or discontinuities on the edges of the original image. This does not work to create horizontally tiled images. Horizontally tiled images are useful as wallpapers, repeating 2d game background art, pictures of repetitive linear objects (like a fence), textures that wrap around a cylinder, or 360 degree panoramas.

**Describe the solution you'd like**
Add some option that permits horizontal tiling (i.e., no discontinuity or seam when placing image next to a copy of itself), but does not enforce any vertical tiling or constraint along the top edge of the image.

Example sketch of a tree which can be horizontally tiled, displayed twice:
![tile-pass](https://user-images.githubusercontent.com/7674289/194440527-95ce8530-b88a-4bc5-be4e-9b83f6aaef29.png)
Example sketch of a tree which can not be horizontally tiled, displayed twice:
![tile-fail](https://user-images.githubusercontent.com/7674289/194440531-7cc4d0df-16c4-4441-b19f-23dca010276f.png)

**Describe alternatives you've considered**
To do this by hand, one could: Render an image, split it in half along a vertical line, swap the two halves, and use inpainting to fix up the interface between the two halves. However, this requires a some luck with the original image, because some seams have no good fix. For example, if the original image shows e.g. half a cat on the left side and half a car on the right side; or if the original image is not horizontal/has the horizon at an angle. I imagine that using a similar padding trick as is done to implement seamless tiling would be much more efficient than this approach.

Having a more general set of tiling options might be useful -- like vertical tiling, or "mirror" tiling where the image must have no seam when placed next to a mirrored copy of itself -- but I don't know if such options would be used very often.

**Additional context**
N/A.

**Comment by lstein** [comment]
This is great. Thank you! Would you be willing to convert this into a pull request so that we can properly track your contribution and credit you? Instructions are here: 

~~~
Anyone who wishes to contribute to this project, whether documentation, features, bug fixes, code cleanup, testing, or code reviews, is very much encouraged to do so. If you are unfamiliar with how to contribute to GitHub projects, here is a [Getting Started Guide](https://opensource.com/article/19/7/create-pull-request-github).

A full set of contribution guidelines, along with templates, are in progress, but for now the most important thing is to make your pull request against the "development" branch, and not against "main". This will help keep public breakage to a minimum and will allow you to propose more radical changes.
~~~

**Comment by psychedelicious** [comment]
Hi @mstoeckl , this was recently implemented in #1132, though the implementation was drawn from another repo. Thanks for your work on the feature anyways.
